indy_node/test/request_handlers/test_context_handler.py

```diff
@@ -0,0 +1,267 @@
+    assert "2http:/..@#$" in str(e.value)
+
+
+def test_validate_context_fail_context_not_uri_or_array_or_object():
+    context = 52
+    with pytest.raises(Exception) as e:
+        validate_context(context)
+    assert "'@context' value must be url, array, or object" in str(e.value)
+
+
+def test_validate_context_pass_value_is_dict():
+    context = {
+        "favoriteColor": "https://example.com/vocab#favoriteColor"
+    }
+    validate_context(context)
+
+
+def test_validate_context_pass_value_is_list_with_dict_and_uri():
+    context = [
+        {
+            "favoriteColor": "https://example.com/vocab#favoriteColor"
+        },
+        "https://www.w3.org/ns/odrl.jsonld"
+    ]
+    validate_context(context)
+
+
+def test_validate_context_pass_context_w3c_example_15():
+    input_dict = {
+        "@context": {
+            "referenceNumber": "https://example.com/vocab#referenceNumber",
+            "favoriteFood": "https://example.com/vocab#favoriteFood"
+        }
+    }
+    validate_context(input_dict)
+
+
+def test_validate_context_fail_value_is_list_with_dict_and_bad_uri():
+    context = [
+        {
+            "favoriteColor": "https://example.com/vocab#favoriteColor"
+        },
+        "this is a bad uri"
+    ]
+    with pytest.raises(Exception) as e:
+        validate_context(context)
+    assert "this is a bad uri" in str(e.value)
+
+
+def test_validate_context_pass_context_w3c_base():
+    # pasted directly out of the reference file, without any format changes
+    # change true to True to correct for python
+    # Sample from specification: https://w3c.github.io/vc-data-model/#base-context
+    # Actual file contents from: https://www.w3.org/2018/credentials/v1
+    validate_context(W3C_BASE_CONTEXT)
+
+
+def test_static_validation_pass_valid_transaction():
+    operation = {
+        "meta": {
+            "name": "TestContext",
+            "version": 1,
+            "type": CONTEXT_TYPE
+        },
+        "data": W3C_BASE_CONTEXT,
+        "type": "200"
+    }
+    req = Request("test", 1, operation, "sig",)
+    ch = ContextHandler(None, None)
+    ch.static_validation(req)
+
+
+def test_validate_context_pass_context_w3c_examples_v1():
+    # test for https://www.w3.org/2018/credentials/examples/v1
+    validate_context(W3C_EXAMPLE_V1_CONTEXT)
+
+
+def test_static_validation_fail_invalid_type():
+    operation = {
+        "meta": {
+            "type": "context",
+            "name": "TestContext",
+            "version": 1
+        },
+        "data": W3C_BASE_CONTEXT,
+        "type": "201"
+    }
+    req = Request("test", 1, operation, "sig",)
+    ch = ContextHandler(None, None)
+    with pytest.raises(LogicError):
+        ch.static_validation(req)
+
+
+def test_static_validation_fail_no_meta():
+    operation = {
+        "data": W3C_BASE_CONTEXT,
+        "type": "200"
+    }
+    req = Request("test", 1, operation, "sig", )
+    ch = ContextHandler(None, None)
+    with pytest.raises(KeyError) as e:
```

Comment:
I'm not sure that `KeyError` is expected error here. I believe expected error should be `InvalidClientRequest`

Reply:
This test has been moved to `indy_node/test/context/test_send_context.py`  and is called `test_validate_operation_fail_no_meta`
It now expects a `TypeError`